addons/hr_expense/models/hr_expense.py

```diff
@@ -120,22 +154,35 @@ def _compute_state(self):
 
     @api.depends('quantity', 'unit_amount', 'tax_ids', 'currency_id')
     def _compute_amount(self):
-        for expense in self:
-            expense.untaxed_amount = expense.unit_amount * expense.quantity
+        for expense in self.filtered('is_nonzero_product_selected'):
+         #   expense.untaxed_amount = expense.unit_amount * expense.quantity
             taxes = expense.tax_ids.compute_all(expense.unit_amount, expense.currency_id, expense.quantity, expense.product_id, expense.employee_id.user_id.partner_id)
             expense.total_amount = taxes.get('total_included')
 
-    @api.depends('date', 'total_amount', 'company_currency_id')
-    def _compute_total_amount_company(self):
+    @api.depends('date', 'total_amount', 'currency_id') # Because currency_id can be changed, company_currency_id - not
+    def _compute_total_amount_company(self):  
         for expense in self:
             amount = 0
-            if expense.company_currency_id:
+            if expense.is_currency_id_different: 
                 date_expense = expense.date
                 amount = expense.currency_id._convert(
                     expense.total_amount, expense.company_currency_id,
                     expense.company_id, date_expense or fields.Date.today())
+            else:
+                amount = expense.total_amount
             expense.total_amount_company = amount
 
+    @api.depends('date', 'total_amount', 'currency_id')
+    def _compute_convert_rate_txt(self):
+        records_with_diff_currency = self.filtered('is_currency_id_different')
+        (self - records_with_diff_currency).convert_rate_txt = False
+        for expense in records_with_diff_currency:
+            date_expense = expense.date
+            rate = expense.currency_id._get_conversion_rate(
+                expense.currency_id, expense.company_currency_id, expense.company_id, date_expense or fields.Date.today())
+            rate_txt = '1 {} = {:.{prec}f} {}'.format(expense.currency_id.name, rate, expense.company_currency_id.name, prec=2)
```

Comment:
```suggestion
            rate_txt = _('1 %(exp_cur)s = %(rate)s %(comp_cur)s', exp_cur=expense.currency_id.name, rate=rate, comp_cur=float_repr(expense.company_currency_id.name, expense.company_currency_id.decimal_places))
```

The method `float_repr` should be imported from `odoo.tools`